src/modalities/config/component_factory.py

```diff
@@ -30,13 +30,32 @@ def build_components(self, config_dict: dict, components_model_type: Type[BaseMo
         Returns:
             BaseModelChild: Instance of the components_model_type with the built components.
         """
-        component_names = list(components_model_type.model_fields.keys())
-        component_dict = self._build_config(config_dict=config_dict, component_names=component_names)
+        # the componentsinstantiaton model allows for the definition of required and optional top-level components
+        # for example the mfu_calculator might not always be required.
+        component_names_required = [
```

Comment:
component_names = dict(required=[], optional=[])

for name, field in components_model_type.model_fields.items():
    (component_names["required"] if field.is_required() else component_names["optional"]).append(name)